pkg/trafficpolicy/trafficpolicy.go

```diff
@@ -189,18 +204,35 @@ func mergeRoutesWeightedClusters(originalRoutes, latestRoutes []*RouteWeightedCl
 	return originalRoutes
 }
 
-// subset returns true if the second array is completely contained in the first array
-func subset(first, second []string) bool {
-	set := make(map[string]bool)
-	for _, value := range first {
-		set[value] = true
+// subset union returns true if either of the slices is a subset of the other and returns a unified slice
+func subsetUnion(first, second []string) (bool, []string) {
+	areSubsets := false
+	unifiedHosts := []string{}
```

Comment:
Function is generic to string slices, but the variable name `unifiedHosts` is specific to hostnames.

Reply:
good catch, I have renamed the variable to be more generic